### PR TITLE
adds regexp support

### PR DESCRIPTION
**Pull request by smnbbrv**

### CODE REVIEW

**Comment by smnbbrv** [comment]
hi @dave-kennedy 

could you please have a look at the pull request? I really need this feature...

**Comment by dave-kennedy** [comment]
I just got a chance to look. Will leave some feedback.

File: index.js [hunk 1]
```diff
@@ -172,11 +172,19 @@ function shouldRemove(node) {
...
         return isEmpty(node);
     }
 
-    return options['remove-tags'].indexOf(node.name) != -1;
+    return isListedInOptions('remove-tags', node.name);
+}
+
+function isListedInOptions(optionsArrayName, name) {
+    var matches = options[optionsArrayName].filter(function(option) {
+        return typeof option !== 'string' && option.test(name) || option === name;
```
Comment (dave-kennedy) [inline index.js:184]: Why not `option instanceof RegExp`?

**Review by dave-kennedy — COMMENTED** [review]
This PR is perfect. I just have one question.

File: index.js (hunk 1 above)
Comment (smnbbrv) [reply index.js:184]: good question. It does not really make that big difference here though... If you think it is important I can of course adapt it

Comment (dave-kennedy) [reply index.js:184]: I think that would be better. If it's not a string, `option.test` might throw an exception. Not that anything else is supported, but it seems more correct.

No worries, I'll get it.

**Pull request merged**